src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java

```diff
@@ -852,4 +857,32 @@ public void cancelOnlyRelevantQueueItem() throws Exception {
         r.assertBuildStatus(Result.ABORTED, r.waitForCompletion(b));
         r.assertLogContains("ran on special agent", b);
     }
+
+    @Test
+    public void garbageCollection() throws Exception {
+        // Pod exists, need to kill the build, delete the agent without deleting the pod.
+        // Wait for the timeout to expire and check that the pod is deleted.
+        var garbageCollection = new GarbageCollection();
+        garbageCollection.setTimeout(120);
```

Comment:
So this test is going to hang for 2m? Can we make it run any faster?

Reply:
Possible, would need to:
* Tweak Async periodic works to run more frequently for tests
* Bypass the current minimum timeout